### PR TITLE
PIM-9470: [Backport] API-1212: Add filters on family search + API-1247: Add filter on category codes + API-1251: Add ability to get categories filtered by parent

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by pierrejolly — APPROVED** [review]

**Review by pchasle — APPROVED** [review]

**Review by pierrejolly — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ExternalApi/CategoryRepository.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            return;
+        }
+
+        $validator = Validation::createValidator();
+        $constraints = [
+            'code' => new Assert\All([
+                new Assert\Collection([
+                    'operator' => new Assert\IdenticalTo([
+                        'value' => 'IN',
+                        'message' => 'In order to search on category codes you must use "IN" operator, {{ compared_value }} given.',
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ExternalApi/CategoryRepository.php:151]:
```suggestion
                        'message' => 'In order to search on category codes you must use "IN" operator, {{ value }} given.',
```

Comment (tseho) [reply src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ExternalApi/CategoryRepository.php:151]: see https://symfony.com/doc/current/reference/constraints/IdenticalTo.html#message

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ExternalApi/CategoryRepository.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                            new Assert\Type('string')
+                        ])
+                    ],
+                ])
+            ]),
+            'parent' => new Assert\All([
+                new Assert\Collection([
+                    'operator' => new Assert\IdenticalTo([
+                        'value' => '=',
+                        'message' => 'In order to search on category parent you must use "=" operator, {{ compared_value }} given.',
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Repository/ExternalApi/CategoryRepository.php:168]: same here

File: src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/FamilyRepository.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            return;
+        }
+
+        $validator = Validation::createValidator();
+        $constraints = [
+            'code' => new Assert\All([
+                new Assert\Collection([
+                    'operator' => new Assert\IdenticalTo([
+                        'value' => 'IN',
+                        'message' => 'In order to search on family codes you must use "IN" operator, {{ compared_value }} given.',
```
Comment (tseho) [inline src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/FamilyRepository.php:130]: same

File: src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/FamilyRepository.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                            new Assert\Type('string')
+                        ])
+                    ],
+                ])
+            ]),
+            'updated' => new Assert\All([
+                new Assert\Collection([
+                    'operator' => new Assert\IdenticalTo([
+                        'value' => '>',
+                        'message' => 'Searching on the "updated" property require the ">" (greater than) operator, {{ compared_value }} given.',
```
Comment (tseho) [inline src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/ExternalApi/FamilyRepository.php:147]: same

File: src/Akeneo/Tool/Bundle/ApiBundle/Resources/config/repositories.yml [hunk 5]
```diff
@@ -23,6 +23,13 @@ services:
             - '%pim_catalog.entity.attribute_option.class%'
             - '@pim_catalog.repository.attribute_option'
 
+    pim_api.repository.attribute_group:
+        class: '%pim_api.repository.api_resource.class%'
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/ApiBundle/Resources/config/repositories.yml:27]:
```suggestion
        class: 'Akeneo\Pim\Structure\Bundle\Doctrine\ORM\Repository\ExternalApi\AttributeGroupRepository'
```

**Pull request merged**